src/raqm.c

```diff
@@ -728,6 +767,432 @@ _raqm_hb_dir (raqm_t *rq, FriBidiLevel level)
   return dir;
 }
 
+enum break_class
+{
+	// input types
+	OP = 0,	// open
+	CL,	// closing punctuation
+	CP,	// closing parentheses (from 5.2.0) (before 5.2.0 treat like CL)
+	QU,	// quotation
+	GL,	// glue
+	NS,	// no-start
+	EX,	// exclamation/interrogation
+	SY,	// Syntax (slash)
+	IS,	// infix (numeric) separator
+	PR,	// prefix
+	PO,	// postfix
+	NU,	// numeric
+	AL,	// alphabetic
+	ID,	// ideograph (atomic)
+	IN,	// inseparable
+	HY,	// hyphen
+	BA,	// break after
+	BB,	// break before
+	B2,	// break both
+	ZW,	// ZW space
+	CM,	// combining mark
+	WJ, // word joiner
+
+	// used for Korean Syllable Block pair table
+	H2, // Hamgul 2 Jamo Syllable
+	H3, // Hangul 3 Jamo Syllable
+	JL, // Jamo leading consonant
+	JV, // Jamo vowel
+	JT, // Jamo trailing consonant
+
+	// these are not handled in the pair tables
+	SA, // South (East) Asian
+	SP,	// space
+	PS,	// paragraph and line separators
+	BK,	// hard break (newline)
+	CR, // carriage return
+	LF, // line feed
+	NL, // next line
+	CB, // contingent break opportunity
+	SG, // surrogate
+	AI, // ambiguous
+	XX, // unknown
```

Comment:
I’m not comfortable with those two letter enums, you should prefix them with `RAQM_BREAK_CLASS_` or something like that.